### PR TITLE
Support Dedalus rules that send results to the next tick

### PR DESCRIPTION
**Pull request by shadaj**

### CODE REVIEW

File: hydroflow/tests/datalog_frontend.rs [hunk 1]
```diff
@@ -1,8 +1,37 @@
-use std::thread;
+use std::{cell::RefCell, rc::Rc, thread};
 
 use hydroflow::futures::StreamExt;
 use hydroflow_datalog::datalog;
 
+struct FakeOutputChannel<T> {
```
Comment (tylerhou) [inline hydroflow/tests/datalog_frontend.rs:6]: `BufferedOutputChannel`?

**Review by tylerhou — APPROVED** [review]

File: hydroflow/tests/datalog_frontend.rs [hunk 2]
```diff
@@ -463,3 +492,34 @@ pub async fn test_join_multiple_then_relation() {
     assert_eq!(out_recv.next().await.unwrap(), (1, 2, 4, 5));
     assert_eq!(out_recv.next().await, None);
 }
+
+#[test]
+pub fn test_next_tick() {
+    let (ints_1_send, ints_1) = hydroflow::util::unbounded_channel::<(usize,)>();
+    let (ints_2_send, ints_2) = hydroflow::util::unbounded_channel::<(usize,)>();
+    let (result, mut result_recv) = fake_output_channel::<(usize,)>();
```
Comment (MingweiSamuel) [inline hydroflow/tests/datalog_frontend.rs:500]: For other tests we've been using `unbounded_channel` and `hydroflow::util::collect_ready`

```rust
    use hydroflow::util::collect_ready;
    let (result, mut result_recv) = hydroflow::util::unbounded_channel::<(usize,)>();
    // ...
    assert_eq!(&[(1,), (2,)], &*collect_ready::<Vec<_>, _>(&mut result_recv));
```

File: hydroflow_datalog_core/src/grammar.rs [hunk 3]
```diff
@@ -32,6 +31,13 @@ pub mod datalog {
         _dot: Option<()>,
     }
 
+    #[derive(Debug, Clone)]
+    pub enum RuleType {
+        Sync(#[rust_sitter::leaf(text = ":-")] ()),
+        NextTick(#[rust_sitter::leaf(text = ":+")] ()),
+        Async(#[rust_sitter::leaf(text = ":~")] ()),
```
Comment (MingweiSamuel) [inline hydroflow_datalog_core/src/grammar.rs:38]: This is a new bloom-inspired syntax?

**Review by MingweiSamuel — APPROVED** [review]

File: hydroflow_datalog_core/src/grammar.rs (hunk 3 above)
Comment (shadaj) [reply hydroflow_datalog_core/src/grammar.rs:38]: Yep! Much simpler than having to analyze the flow of `t` through `succ`/`choose`.

**Pull request merged**